custom/icds_reports/utils/aggregation_helpers/distributed/primary_private_school.py

```diff
@@ -30,24 +30,21 @@ def aggregation_query(self):
           date_admission_private_school, returned_private_school, date_return_private_school,
           admitted_primary_school, date_admission_primary_school
         ) (
-          SELECT DISTINCT
+        SELECT DISTINCT ON (person_case_id)
             %(state_id)s AS state_id,
-            LAST_VALUE(supervisor_id) over w as supervisor_id,
+            supervisor_id,
             %(month)s::DATE AS month,
-            LAST_VALUE(person_case_id) over w AS person_case_id,
-            LAST_VALUE(admitted_private_school) over w ='yes' AS admitted_private_school,
-            LAST_VALUE(date_admission_private_school) over w  AS date_admission_private_school,
-            LAST_VALUE(returned_private_school) over w='yes'  AS returned_private_school,
-            LAST_VALUE(date_return_private_school) over w  AS date_return_private_school,
-            CLAST_VALUE(admitted_primary_school) over w='yes'  AS admitted_primary_school,
-            LAST_VALUE(date_admission_primary_school) over w AS date_admission_primary_school
-          FROM "{ucr_tablename}"
-          WHERE state_id = %(state_id)s AND
+            person_case_id,
+            admitted_private_school='yes' AS admitted_private_school,
+            date_admission_private_school,
+            returned_private_school='yes' AS returned_private_school,
+            date_return_private_school,
+            admitted_primary_school='yes' AS admitted_primary_school,
+            date_admission_primary_school
+            from "{ucr_tablename}"
+            WHERE state_id = %(state_id)s AND
                 timeend < %(next_month_start)s
```

Comment:
This query looks at all forms of this type ever submitted. That will be very slow as the number increases even now is likely quite slow. It should only have to look at forms from the current month. If there needs to be carried over data, you can see how other intermediate tables do that by joining with the previous month (like the ccs ones).

Reply:
Hey, I had thought of that. But there is no past table for this therefore that query will give the wrong result. I can think of two solutions.
1) leave it like this for this release and next month update the query to the full join thing as other places do them.
2) do that change now and after that is deployed and before its aggregated. create the last month table with the current query which will be one time and see all forms.
@calellowitz

Reply:
@calellowitz  I am looking for your opinion on this.